packages/queued-request-controller/tests/QueuedRequestMiddleware.test.ts

```diff
@@ -0,0 +1,323 @@
+    'NetworkController:setProviderType',
+    mockSetProviderType,
+  );
+
+  const mockSetActiveNetwork = jest.fn().mockResolvedValue(true);
+  messenger.registerActionHandler(
+    'NetworkController:setActiveNetwork',
+    mockSetActiveNetwork,
+  );
+
+  const mockSetNetworkClientIdForDomain = jest.fn().mockResolvedValue(true);
+  messenger.registerActionHandler(
+    SelectedNetworkControllerActionTypes.setNetworkClientIdForDomain,
+    mockSetNetworkClientIdForDomain,
+  );
+
+  return {
+    getProviderConfig: mockGetProviderConfig,
+    getNetworkClientById: mockGetNetworkClientById,
+    enqueueRequest: mockEnqueueRequest,
+    addRequest: mockAddRequest,
+    setActiveNetwork: mockSetActiveNetwork,
+    setProviderType: mockSetProviderType,
+    setNetworkClientIdForDomain: mockSetNetworkClientIdForDomain,
+  };
+};
+
+const noop = jest.fn();
+
+describe('createQueuedRequesMitddleware', () => {
+  it('should not enqueue the request when useRequestQueue is false', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware(messenger, () => false);
+    const mocks = buildMocks(messenger);
+
+    const req = {
+      origin: 'example.com',
+    } as any;
+
+    await new Promise((resolve) => middleware(req, {} as any, resolve, noop));
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  it('should not enqueue the request when there is no confirmation', async () => {
+    const messenger = buildMessenger();
+    const middleware = createQueuedRequestMiddleware(messenger, () => true);
+    const mocks = buildMocks(messenger);
+
+    const req = {
+      origin: 'example.com',
+      method: 'eth_chainId',
+    } as any;
+
+    await new Promise((resolve) => middleware(req, {} as any, resolve, noop));
+
+    expect(mocks.enqueueRequest).not.toHaveBeenCalled();
+  });
+
+  describe('confirmations', () => {
+    it('should resolve requests that require confirmations', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware(messenger, () => true);
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        origin: 'example.com',
+        method: 'eth_sendTransaction',
+      } as any;
+
+      await new Promise((resolve) => middleware(req, {} as any, resolve, noop));
+
+      expect(mocks.addRequest).not.toHaveBeenCalled();
+      expect(mocks.enqueueRequest).toHaveBeenCalled();
+      expect(mocks.getNetworkClientById).toHaveBeenCalled();
+      expect(mocks.getProviderConfig).toHaveBeenCalled();
+    });
+
+    it('switchEthereumChain calls get queued but we dont check the current network', async () => {
+      const messenger = buildMessenger();
+      const middleware = createQueuedRequestMiddleware(messenger, () => true);
+      const mocks = buildMocks(messenger);
+
+      const req = {
+        origin: 'example.com',
+        method: 'wallet_switchEthereumChain',
+      } as any;
```

Comment:
if possible, i believe avoid `any` even in specs is ideal

Reply:
Yes please :)